oci/oci.go

```diff
@@ -1 +1,91 @@
 package oci
+
+import (
+	"github.com/pkg/errors"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+// annotation spex from https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys
+const (
+	separator = `(?:[-._:@+]|--)`
+	alphanum  = `(?:[A-Za-z0-9]+)`
+	component = `(?:` + alphanum + `(?:` + separator + alphanum + `)*)`
+)
+
+var refRegexp = regexp.MustCompile(`^` + component + `(?:/` + component + `)*$`)
+var windowsRefRegexp = regexp.MustCompile(`^([a-zA-Z]:\\.+?):(.*)$`)
+
+// ValidImageName returns true is the specified image matches the open-containers image name specs, false otherwise.
+func ValidImageName(image string) bool {
+	return refRegexp.MatchString(image)
+}
+
+// SplitDirAndImage tries to split the provided OCI reference into the OCI directory and image.
+// The image is returned as string pointer, allowing for nil being returned, if no image part could be extraced from the reference.
+// Neither directory nor image parts are validated at this stage.
+func SplitDirAndImage(reference string) (string, *string) {
+	if runtime.GOOS == "windows" {
+		return splitDirAndImageWindows(reference)
+	}
+	return splitDirAndImageNonWindows(reference)
+}
+
+func splitDirAndImageWindows(reference string) (string, *string) {
+	groups := windowsRefRegexp.FindStringSubmatch(reference)
+	// nil group means no match
+	if groups == nil {
+		return reference, nil
+	}
+
+	// we expect three elements. First one full match, second the capture group for the dir and
+	// the third the capture group for the image
+	if len(groups) != 3 {
+		return reference, nil
+	}
+	return groups[1], &groups[2]
+}
+
+func splitDirAndImageNonWindows(reference string) (string, *string) {
+	sep := strings.SplitN(reference, ":", 2)
+	dir := sep[0]
+
+	var image *string
+	if len(sep) == 2 {
+		image = &sep[1]
+	}
+	return dir, image
+}
+
+// ValidateScope takes the OCI directory and validates its scope.
+func ValidateScope(scope string, dir string) error {
+	if runtime.GOOS == "windows" {
+		return validateScopeWindows(scope, dir)
+	}
+	return validateScopeNonWindows(scope, dir)
```

Comment:
Can the two variants share code via `filepath.IsAbs`? (That one also handles UNC paths, but that seems to be compatible enough with our formats.)

Reply:
Potentially. I guess you are talking about just calling "filepath.IsAbs" w/o a OS specific switch. There would still be 

    if scope == "/" {
        return errors.New(`Invalid scope "/": Use the generic default scope ""`)
    }

to take care of which is non Windows specific (unless you suggest we drop this).

Another disadvantage of using filepath.IsAbs is that one cannot easily write tests for Windows paths. In the current code I can test _validateScopeWindows_ even w/o running the test suite on Windows. Once filepath.IsAbs is used that is not the case anymore. 

I am undecided. I see pro and cons for both and tempted to prefer the current code to start with.

Reply:
Sure, the non-Windows testing is a valid concern. I suppose we don’t need to care about UNC paths at least for now.